presto-native-execution/presto_cpp/main/TaskManager.cpp

```diff
@@ -42,30 +42,37 @@ namespace {
 // the file system and sets the path to it to the Task.
 static void maybeSetupTaskSpillDirectory(
     const core::PlanFragment& planFragment,
-    exec::Task& execTask,
+    exec::Task& veloxTask,
     const std::string& baseSpillDirectory) {
   if (baseSpillDirectory.empty() ||
-      !planFragment.canSpill(execTask.queryCtx()->queryConfig())) {
+      !planFragment.canSpill(veloxTask.queryCtx()->queryConfig())) {
     return;
   }
 
   const auto includeNodeInSpillPath =
       SystemConfig::instance()->includeNodeInSpillPath();
   auto nodeConfig = NodeConfig::instance();
-
   const auto taskSpillDirPath = TaskManager::buildTaskSpillDirectoryPath(
       baseSpillDirectory,
       nodeConfig->nodeInternalAddress(),
       nodeConfig->nodeId(),
-      execTask.queryCtx()->queryId(),
-      execTask.taskId(),
+      veloxTask.queryCtx()->queryId(),
+      veloxTask.taskId(),
       includeNodeInSpillPath);
-  execTask.setSpillDirectory(taskSpillDirPath);
-  // Create folder for the task spilling.
-  auto fileSystem =
-      velox::filesystems::getFileSystem(taskSpillDirPath, nullptr);
-  VELOX_CHECK_NOT_NULL(fileSystem, "File System is null!");
-  fileSystem->mkdir(taskSpillDirPath);
+
+  try {
+    // Create folder for the task spilling.
+    auto fileSystem =
+        velox::filesystems::getFileSystem(taskSpillDirPath, nullptr);
+    VELOX_CHECK_NOT_NULL(fileSystem, "File System is null!");
+    fileSystem->mkdir(taskSpillDirPath);
+  } catch (...) {
+    LOG(WARNING) << "Failed to create spill directory for velox task "
+                 << veloxTask.taskId() << ". Error: "
+                 << folly::exceptionStr(std::current_exception());
+    return;
+  }
+  veloxTask.setSpillDirectory(taskSpillDirPath);
```

Comment:
Thanks, curious what will happen when path creation fails and we still set the path.

Reply:
It will get another exception on rmdir and might distract the debug but the query fails so it won't get executed.